sklearn/tree/_tree.pyx

```diff
@@ -438,7 +438,7 @@ cdef class BestFirstTreeBuilder(TreeBuilder):
         n_node_samples = end - start
         is_leaf = ((depth > self.max_depth) or
                    (n_node_samples < self.min_samples_split) or
-                   (n_node_samples < 2 * self.min_samples_leaf) or
+                   (n_node_samples < self.min_samples_leaf) or
```

Comment:
Ah, sorry for my incomprehension. Yeah, I agree with you on that count ; the condition is never true and so the Splitter manages the condition. I'll keep looking into the code for a bit and verify I understand what's going on.  I feel like there's something off somewhere in either `min_weight_fraction_leaf `or `min_samples_leaf` in order for it not to be working out of the box, no?